libgnucash/engine/AccountP.hpp

```diff
@@ -113,8 +113,10 @@ typedef struct AccountPrivate
     gnc_numeric cleared_balance;
     gnc_numeric reconciled_balance;
 
-    std::optional<gnc_numeric> higher_balance_limit;
-    std::optional<gnc_numeric> lower_balance_limit;
+    // outer optional to denote if balance_limit is cached
```

Comment:
Too clever by half. Change your design so that the limits are loaded from KVP at object initialization and save the value to KVP in xaccAccountCommitEdit.

Reply:
> Too clever by half. Change your design so that the limits are loaded from KVP at object initialization and save the value to KVP in xaccAccountCommitEdit.

Not too sure about that. `gnc_account_init` shouldn't assume a book is available to read kvp from. I'm thinking a 'balance_limits_are_cached' bool may be used. Or, forego caching them altogether.

Reply:
What? You can't create any other kind of QofInstance without a book, not that that has anything to do with this.

You have the wrong mental model of KVP. KVP is a storage extension to provide class member variables that don't require a matching XML element name and therefore require adding handler code to the backend. Ideally KVP wouldn't exist and we'd maintain and version datafiles, but we don't. Next best would have the KVP code in the backends and map directly to instance member variables so the engine wouldn't have to worry about them.

Reply:
> The last issue is the optional<optional> which I admit is confusing but I'm not sure where to initialise the balance limits except on the first getter call.

Init them in gnc_account_init. Slots go with the QofInstance, and that's part of struct Account. It has nothing to do with the book.

Or don't cache them at all. Retrieving a KVP isn't that expensive.